mantidproject/mantid
# Reflectometry GUI Restructuring 15: disable editing keys

***PR ~~#25897~~ should be merged first.*** In the meantime this can be reviewed by comparing against [that branch](https://github.com/gemmaguest/mantid/compare/23027_refl_gui_catalog_login...23027_refl_gui_disable_editing_keys)

**Description of work.**

Ongoing work to integrate the `JobTreeView` into the `ISIS Reflectometry` GUI. This PR is being merged to a feature branch because not all functionality is implemented yet.

This PR disables editing keys while processing is in progress. Editing the table during processing can cause crashing.

**To test:**

This should be tested by someone at ISIS.

- Open the `ISIS Reflectometry` interface
- Set the instrument to `INTER` and enter the investigation ID `1120015` and click `Search`. Select all of the resulting rows and click `Transfer` to copy them to the main table.
- Check that the following keys work:
  - Enter will move to the next row/group in the same level or add a new row/group if at the last one
  - Ctrl-Enter will add a new row to the selected group. It will do nothing if you have a row selected.
  - Shift-Enter will move to editing the cell above
  - Ctrl-X, Ctrl-C and Ctrl-V will cut/copy/paste
  - Delete will delete a row/group
  - F2 will edit a cell and tab/shift-tab will move to editing the next/previous cell. If you reach the end of a row it will progress to the first cell in the next row and if there is no next row it will insert a new row.
- Deselect everything and click `Process`. Check that the keys now have no effect while processing is in progress and you cannot cause a crash.

Refs #23027

*This does not require release notes* because **it re-implements existing functionality**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
